moveit_core/robot_state/test/robot_state_test.cpp

```diff
@@ -46,34 +46,49 @@
 namespace
 {
 constexpr double EPSILON{ 1.e-9 };
-}
 
-#if 0  // unused function
-static bool sameStringIgnoringWS(const std::string& s1, const std::string& s2)
+// Helper to create an Eigen::VectorXd from an initializer list, e.g.:
+//  auto vector = makeVector({1.0, 2.0, 3.0});
+Eigen::VectorXd makeVector(const std::vector<double>& values)
 {
-  unsigned int i1 = 0;
-  unsigned int i2 = 0;
-  while (i1 < s1.size() && isspace(s1[i1]))
-    i1++;
-  while (i2 < s2.size() && isspace(s2[i2]))
-    i2++;
-  while (i1 < s1.size() && i2 < s2.size())
+  Eigen::VectorXd vector = Eigen::VectorXd::Zero(values.size());
+  for (std::size_t i = 0; i < values.size(); i++)
   {
-    if (i1 < s1.size() && i2 < s2.size())
-    {
-      if (s1[i1] != s2[i2])
-        return false;
-      i1++;
-      i2++;
-    }
-    while (i1 < s1.size() && isspace(s1[i1]))
-      i1++;
-    while (i2 < s2.size() && isspace(s2[i2]))
-      i2++;
+    vector[i] = values[i];
   }
-  return i1 == s1.size() && i2 == s2.size();
+  return vector;
 }
-#endif
+
+// Checks the validity of state->getJacobian() at the given 'joint_values' and 'joint_velocities'.
+void CheckJacobian(moveit::core::RobotState* state, const moveit::core::JointModelGroup& joint_model_group,
+                   const Eigen::VectorXd& joint_values, const Eigen::VectorXd& joint_velocities)
+{
+  ASSERT_TRUE(state != nullptr);
+
+  // Using the Jacobian, compute the Cartesian velocity vector at which the end-effector would move, with the given
+  // joint velocities.
+  state->setJointGroupPositions(&joint_model_group, joint_values);
```

Comment:
not sure if this applies for the implemented tests, but prepending this with `state->setToDefaultPositions()` could prevent side effects.